### ISSUE TITLE
[FP] Foreach can probably be used: index decoupled from array

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1258]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), v2.0.14 |

Current behaviour
--
I'm not sure how the detection is done but this code triggers the inspection
```php
    $nb=2;
    $t_numbers = [0,1,2,3,4,5];
    for($i=0;$i<$nb;$i++)
    {
        echo $t_numbers[$i];
    }
```

But $nb limits the loops to 2. If we replace by foreach, we will loop 6 times. I don't think this inspection should be triggered if $nb is not assigned with a variation of count($t_numbers)

Thanks

**Comment by kalessil** [comment]
Fixed!

**Comment by orklah** [comment]
Thanks!

**Comment by orklah** [comment]
In your commit, you used the "count" function hardcoded in "isLimitFor". You may want to add sizeof as well to match more cases...

**Comment by kalessil** [comment]
It's an alias for count, which we are marking as deprecated, so I don't see value in supporting "sizeof".

**Comment by orklah** [comment]
This inspection is about performance and readability. The sizeof one is just code style. I think people tend to fix performance before code style in legacy code. 

And then, there may be people who prefer using sizeof anyway (C habits?) who won't ever see this inspection because of that

**Comment by kalessil** [comment]
Sounds good to me: reward for respecting the plugin hints.

This developers group is not a target one, correcting old habits will bring new value - IMO that's a fair deal.

**Comment by orklah** [comment]
Fair enough :)

I was just making sure this was not something you forgot :)

Thanks again for the fix!

**Comment by orklah** [comment]
FYI, I kinda expected to see less cases after this issue was resolved, but since the last release of EA, I actually see a lot more cases. Cool :)
